File: src/main/java/ar/edu/unlam/tallerweb1/delivery/api/ApiShowBalance.java
package ar.edu.unlam.tallerweb1.delivery.api;

import ar.edu.unlam.tallerweb1.delivery.Controller;
import ar.edu.unlam.tallerweb1.domain.FriendsGroupService;

import javax.servlet.http.HttpServletRequest;
import java.util.Map;

public class ApiShowBalance implements Controller {
    private FriendsGroupService friendsGroupService;

    public ApiShowBalance(FriendsGroupService friendsGroupService) {
        this.friendsGroupService = friendsGroupService;
    }

    @Override
    public Map<String, Double> invoke(HttpServletRequest request) {
        String user = request.getParameter("user");
        return friendsGroupService.getBalance(user);
    }
}

File: src/main/java/ar/edu/unlam/tallerweb1/delivery/api/ApiShowBalance.kt
package ar.edu.unlam.tallerweb1.delivery.api

import ar.edu.unlam.tallerweb1.delivery.Controller
import ar.edu.unlam.tallerweb1.domain.usecases.GetBalance
import javax.servlet.http.HttpServletRequest

class ApiShowBalance(private val getBalance: GetBalance) : Controller {
    override operator fun invoke(request: HttpServletRequest) = getBalance(request.getParameter("user"))
}
